pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -207,6 +218,111 @@ func (a Adapter) Build(parameters common.BuildParameters) (err error) {
 	return
 }
 
+func determinePort(parameters common.DeployParameters) string {
+	// TODO: Determine port to use (from env.yaml or other location!!)
+	deploymentPort := ""
+	for _, localURL := range parameters.EnvSpecificInfo.GetURL() {
+		if localURL.Kind != envinfo.DOCKER {
+			deploymentPort = strconv.Itoa(localURL.Port)
+			break
+		}
+	}
+	return deploymentPort
+}
```

Comment:
So we are saying that the url needs to do `odo url create` before `odo deploy`, I thing that might be fine. Just need to document it

Reply:
It all depends on how/where we are eventually determining the port info from. At present I just decided to use the port that was setup for use with the push. This will probably change.